### PR TITLE
Added a prop to support custom options in the dropdown.

### PR DESCRIPTION
**Pull request by shishiranshuman**
I have added an example on the demo page as well. @benkeen Request you to have a look and let me know if any improvements are needed.

react_country_region_selector

screen_shot_2018-01-22_at_4 22 41_pm

react_country_region_selector-2

### CODE REVIEW

**Comment by benkeen** [comment]
Hi @shishiranshuman, the code looks terrific, just a couple of things mentioned above.

**Comment by shishiranshuman** [comment]
@benkeen Cool. Thanks. I think you forgot to submit the review.

> just a couple of things mentioned above.

I could not find any comment in the PR.

**Comment by shishiranshuman** [comment]
@benkeen Cool. Thanks. I think you forgot to submit the review.

> just a couple of things mentioned above.

I could not find any comment in the PR.

**Comment by benkeen** [comment]
> @benkeen Cool. Thanks. I think you forgot to submit the review.
> 
> > just a couple of things mentioned above.
> 
> I could not find any comment in the PR.

Thx! Was rushing - missed that step, sorry.

File: src/RegionDropdown.js [hunk 1]
```diff
@@ -20,7 +20,21 @@ export default class RegionDropdown extends Component {
 		if (nextProps.country === this.props.country) {
 			return;
 		}
-		this.setState({ regions: this.getRegions(nextProps.country) });
+	
+		this.setState({ regions: [
+			...this.getRegions(nextProps.country),
+			...this.getCustomOptions()
```
[the remark this answers is not in the document]
Comment (shishiranshuman) [reply src/RegionDropdown.js]: Yeah. I agree. Should have considered this case. Thanks

File: README.md [hunk 2]
```diff
@@ -153,6 +153,7 @@ added to the actual DOM element._
 | valueType | No | `"full"` | `string` | Either `"full"` or `"short"`. This controls the actual `value` attribute of each `<option>` in the dropdown.  |
 | disableWhenEmpty | No | `false` | `boolean` | Disables the region field when the user hasn't selected a country. |
 | disabled | No | `false` | `boolean` | Disables the region field. If set to true, it overrides `disableWhenEmpty` | 
+| customOptions | No | `[]` | `Array<string>` | Adds a list of string to the region dropdown.
```
[the remark this answers is not in the document]
Comment (shishiranshuman) [reply README.md]: Done

**Comment by shishiranshuman** [comment]
@benkeen I have done the changes. Please have a look.

File: src/tests/RegionDropdown.test.js [hunk 3]
```diff
@@ -176,6 +176,31 @@ describe('RegionDropdown', () => {
...
+						showDefaultOption={false}
+						customOptions={['All']} />
+				);
+				
+				wrapper.setProps({country: 'Antarctica'});
+				expect(wrapper.find('select').childAt(1).text()).toBe('All');
+			});
+
+			it('should throw an error when the duplicate values are present.', () => {
+				console.error = jest.fn();
```
Comment (benkeen) [inline src/tests/RegionDropdown.test.js]: Nice!

**Comment by benkeen** [comment]
Great work. thanks @shishiranshuman! I'll merge now and this'll be included in the next version.

**Pull request merged**